pennylane_lightning/core/src/gates/GateOperation.hpp

```diff
@@ -109,4 +152,12 @@ enum class MatrixOperation : uint32_t {
     /* END (placeholder) */
     END
 };
+
+enum class ControlledMatrixOperation : uint32_t {
```

Comment:
Do we expect any additional entries in this enum? I wonder if it is overkill if the single entry will always be `NCMultiQubitOp`.

Reply:
The main reason this is here is that the dynamic dispatcher + kernels architecture uses these enums. I started with only `NCMultiQubitOp` and nothing else (no gate ops, no gens) because the original ticket only had a title like "Support arbitrary controlled ops". So that was the quickest way to satisfy that spec, but the scope significantly enlarged since then. I think we could add 1- and 2-qubit `NCQubitOp`s by analogy with the non-controlled gates.